azavea/raster-foundry
# Add Dropbox Frontend views

## Overview

Add frontend for dropbox connection

### Checklist

~- [ ] Swagger specification updated, if necessary~
~- [ ] Styleguide updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo
![image](https://user-images.githubusercontent.com/4392704/26845719-6f0f19d8-4ac5-11e7-9527-36ac56f4819f.png)

### Notes

API is not implemented yet (see #1918), so all requests are expected to fail

## Testing Instructions

* Make sure you have `DROPBOX_KEY` defined in your .env file
* Verify that app routes still work in html5 mode. The app routed using hashes before, but OAuth doesn't allow url fragments in the callback url.
* Go to http://localhost:9091/settings/connections
* Connect by clicking the dropbox connect button
* Verify that the app tries to post the correct information to the server, and handles failure for both dropbox errors and api errors. To force an error on dropbox, click cancel instead of allow when the app requests permission.
* Verify that when the child window is closed / failed / successful, the app stops polling the child window. This can be done by adding a `this.$log.log('polled')` on `line 69` of `connections.controller.js`
* Verify that the missing fields in the profile view are now available: password reset and feature flags. Feature flags are only visible when in development mode.
* Verify that the frontend acknowledges when the user is already dropbox authenticated. Manually update the field in the database if necessary.

Closes #1835 
Closes #1876
